tasks/components/CastIntentTask.js

```diff
@@ -0,0 +1,74 @@
+import h from 'react-hyperscript'
+import { isNil, merge, keys, takeLast, toString, mapObjIndexed, split, forEach, pick, find, equals, values } from 'ramda'
+import { reduxForm as connectForm, Field } from 'redux-form'
+import { compose } from 'recompose'
+
+import ProductList from '../../ordering/components/ProductList'
+import SingleViewProduct from '../../ordering/components/SingleViewProduct'
+
+
+// import styles from '../styles/CastIntentTask'
+
+function CastIntentTask (props) {
+  const { actions, currentAgent, taskPlan, routerParams } = props
+  if(isNil(currentAgent)) {
+    return null
+  }
+
+  const { productId } = routerParams
+  if (productId) {
+    return h(SingleProduct, props)
+  } else {
+    return h(ManyProducts, props)
+  }
+}
+
+function SingleProduct (props) {
+  const { actions, products, routerParams, currentAgent, taskPlan, orderIntents } = props
+  const product = products[routerParams.productId]
+  const nextProps = merge(props, { product, onSubmit })
+  // TODO pass onNavigate to allow SingleViewProduct
+  // to navigate back to list product view
+  return h(SingleViewProduct, nextProps)
+  function onSubmit (value) {
+
+    const submittedOrderIntents = values(mapObjIndexed((quantity, priceSpecString) => {
+      return {
+        agentId: currentAgent.id,
+        desiredQuantity: parseInt(quantity),
+        productId: product.id,
+        priceSpecId: parseInt(split('-', priceSpecString)[1]),
+        orderId: taskPlan.params.orderId
+      }
+    }, value.priceSpecs))
+
+    forEach((submittedOrderIntent) => {
+      const scopedSubmittedOrderIntent = pick(['orderId', 'priceSpecId', 'productId'], submittedOrderIntent)
```

Comment:
recommend splitting ramda calls into two steps:

1. create the function
2. use the function

so for example, this line could be

```
const getScopedSubmittedOrderIntent = pick(['orderId', 'priceSpec', 'productId'])
const scopedSubmittedOrderIntent = getScopedSubmittedOrderIntent(submittedOrderIntent)
```

and in this case, the first line could be moved to the top-level scope of this file, it doesn't depend on any variables inside this current scope.

[why ramda?](http://fr.umio.us/why-ramda/)
[why curry helps](https://hughfdjackson.com/javascript/why-curry-helps/)